src/data/WorkSchedData.js

```diff
@@ -51,51 +66,102 @@ class Monday extends Component {
+                    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
+                  </ModalBody>
+                  <ModalFooter>
+                    <Button color="primary" onClick={this.toggle}>Do Something</Button>{' '}
+                    <Button color="secondary" onClick={this.toggle}>Cancel</Button>
+                  </ModalFooter>
+            </Modal>
             </div>
         );
     }
 }
 
 class Tuesday extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            modal: false
+        };
+        
+        this.toggle = this.toggle.bind(this);
+    }
+    
+    toggle() {
+        this.setState({
+            modal: !this.state.modal
+        });
+    }
+    
     render() {
         return(
             <div>
                 <ListGroup>
                     <ListGroupItem>Tuesday</ListGroupItem>
                     <ListGroupItem></ListGroupItem>
                     <ListGroupItem></ListGroupItem>
-                    <ListGroupItem className='oneHour rowing'><div>10:00-11:00</div>Rowing Workout</ListGroupItem>
+                    <ListGroupItem className='oneHour rowing' onClick={this.toggle}><div>10:00-11:00</div>Rowing Workout</ListGroupItem>
                     <ListGroupItem></ListGroupItem>
-                    <ListGroupItem className='oneHourHalf restorive'><div>11:30-13:00</div>Restoration Yoga</ListGroupItem>
+                    <ListGroupItem className='oneHourHalf restorive' onClick={this.toggle}><div>11:30-13:00</div>Restoration Yoga</ListGroupItem>
                     <ListGroupItem></ListGroupItem>
-                    <ListGroupItem className='oneHourHalf abs'><div>13:30-15:00</div>Abs Circuit</ListGroupItem>
+                    <ListGroupItem className='oneHourHalf abs' onClick={this.toggle}><div>13:30-15:00</div>Abs Circuit</ListGroupItem>
                     <ListGroupItem></ListGroupItem>
                     <ListGroupItem className='short0'></ListGroupItem>
-                    <ListGroupItem className='oneHour yoga'><div>15:45-16:45</div>Yoga Level 1</ListGroupItem>
+                    <ListGroupItem className='oneHour yoga' onClick={this.toggle}><div>15:45-16:45</div>Yoga Level 1</ListGroupItem>
                     <ListGroupItem className='short0'></ListGroupItem>
                     <ListGroupItem></ListGroupItem>
                     <ListGroupItem></ListGroupItem>
                     <ListGroupItem></ListGroupItem>
                 </ListGroup>
+            <Modal isOpen={this.state.modal} toggle={this.toggle} className={this.props.className}>
```

Comment:
Please make modal as a separated component. Because we have here a duplicated code 3 times

Reply:
@atherdon I have seperated modal components to Modals.js